apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/utils/APIUtil.java

```diff
@@ -2517,6 +2549,51 @@ public static JSONObject executeQueryOnStreamProcessor(String appName, String qu
 
     }
 
+    /**
+     * This method is used to extact group ids from Extractor.
+     *
+     * @param response               login response String.
+     * @param groupingExtractorClass extractor class.
+     * @return group ids
+     * @throws APIManagementException Throws is an error occured when stractoing group Ids
+     */
+    public static String[] getGroupIdsFromExtractor(String response, String groupingExtractorClass)
+            throws APIManagementException {
+
+        if (groupingExtractorClass != null) {
+            try {
+                LoginPostExecutor groupingExtractor = (LoginPostExecutor) APIUtil.getClassInstance
+                        (groupingExtractorClass);
+                //switching 2.1.0 and 2.2.0
+                if (APIUtil.isMultiGroupAppSharingEnabled()) {
+                    NewPostLoginExecutor newGroupIdListExtractor = (NewPostLoginExecutor) groupingExtractor;
+                    return newGroupIdListExtractor.getGroupingIdentifierList(response);
+                } else {
+                    String groupId = groupingExtractor.getGroupingIdentifiers(response);
+                    return new String[]{groupId};
+                }
+
+            } catch (ClassNotFoundException e) {
+                String msg = groupingExtractorClass + " is not found in runtime";
+                log.error(msg, e);
+                throw new APIManagementException(msg, e);
+            } catch (ClassCastException e) {
+                String msg = "Cannot cast " + groupingExtractorClass + " NewPostLoginExecutor";
+                log.error(msg, e);
+                throw new APIManagementException(msg, e);
+            } catch (IllegalAccessException e) {
+                String msg = "Error occurred while invocation of getGroupingIdentifier method";
+                log.error(msg, e);
+                throw new APIManagementException(msg, e);
+            } catch (InstantiationException e) {
+                String msg = "Error occurred while instantiating " + groupingExtractorClass + " class";
+                log.error(msg, e);
+                throw new APIManagementException(msg, e);
```

Comment:
we can use single `throw new APIManagementException(msg, e);` at last, instead of adding each lines